src/panels/lovelace/mixins/element-click-mixin.js

```diff
@@ -10,13 +10,71 @@ import computeStateName from '../../../common/entity/compute_state_name';
  * @appliesMixin NavigateMixin
  */
 export default dedupingMixin(superClass => class extends NavigateMixin(EventsMixin(superClass)) {
-  handleClick(hass, config) {
-    const tapAction = config.tap_action || 'more-info';
+
+  registerMouse(config, enableHold = false) {
+    this.mouseHeld = false;
+    this.mouseHoldTimer = null;
+
+    let reset = () => {
+      clearTimeout(this.mouseHoldTimer);
+      this.mouseHeld = false;
+    };
+
+    let click = () =>  {
+      let held = this.mouseHeld;
+      this.mouseHeld = false;
+      if(held) return;
+      this.handleClick(this.hass, config, false);
+    };
+
+    let down = () => {
+      this.mouseHoldTimer = setTimeout(() => {
+        this.mouseHeld = true;
+        this.handleClick(this.hass, config, true);
+      }, 300);
+    };
+
+    let up = () => {
+      clearTimeout(this.mouseHoldTimer);
+      let moreInfoDialog = document
+        .querySelector('home-assistant')
+        .$$('ha-more-info-dialog');
+      if (moreInfoDialog) {
+        setTimeout(() => moreInfoDialog.removeAttribute('modal'), 0);
```

Comment:
This removes the `modal` attribute from the more-info-dialog described in a comment below.
The use of a timeout ensures that the attribute is not removed until after all `click/mouseup` event handlers have been run.